contracts/token/common/ERC2981/ERC2981Internal.sol

```diff
@@ -19,22 +19,19 @@ abstract contract ERC2981Internal is IERC2981Internal {
     {
         ERC2981Storage.Layout storage l = ERC2981Storage.layout();
 
-        uint256 royalty = l.royalties[tokenId] > 0
-            ? l.royalties[tokenId]
-            : l.royalty;
+        uint256 royalty = _getRoyaltyBPS(l, tokenId);
+        if (royalty > 10000) revert ERC2981Internal__RoyaltyExceedsMax();
 
-        if (0 == royalty) {
-            revert ERC2981Internal__RoyaltyNotSet();
-        }
-
-        if (royalty > 10000) {
-            revert ERC2981Internal__RoyaltyExceedsMax();
-        }
-
-        uint256 royaltyAmount = salePrice > 0
-            ? (royalty * salePrice) / 10000
-            : 0;
+        return (l.receiver, (royalty * salePrice) / 10000);
+    }
 
-        return (l.receiver, royaltyAmount);
+    function _getRoyaltyBPS(ERC2981Storage.Layout storage l, uint256 tokenId)
```

Comment:
Don't pass the storage layout as an argument here.  We did this extensively in the Premia contracts, and I think the added value there is questionable.  It may have been worth it at the time due to contract size savings.  In SolidState, however, we need to place a greater emphasis on clean interfaces, even when it increases gas usage slightly.